File: emccd_detect_m/scripts/plot_gain_curves.m
clear;  format compact; close all; clc;
addpath('../util');

gain = 100;
n_array = 1:6;
x_array = 0:10:2000;
pdf = zeros(length(n_array), length(x_array));
legend_str = {};
for n = n_array
    pdf(n, :) = em_gain_pdf(x_array, n, gain);
    legend_str{end+1} = ['n = ', num2str(n)];  %#ok<SAGROW>
end

figure;
plot(x_array/gain, pdf);
ylabel('probability density');
xlabel('output counts / gain');
legend(legend_str);

function pdf = em_gain_pdf(x, n, g)
pdf = x.^(n-1) .* exp(-x/g) / (g^n * factorial(n-1));
end

File: emccd_detect/scripts/plot_gain_curves.py
from __future__ import absolute_import, division, print_function

import matplotlib.pyplot as plt
import numpy as np


def em_gain_pdf(x, n, g):
    return x**(n-1) * np.exp(-x/g) / (g**n * np.math.factorial(n-1))


gain = 100
n_array = np.arange(1, 7)
x_array = np.arange(0, 2001, 10)
legend_str = []
plt.figure()
for i, n in enumerate(n_array):
    pdf = em_gain_pdf(x_array, n, gain)
    legend_str.append('n = ' + str(n))
    plt.plot(x_array/gain, pdf, linewidth=.5)

plt.xlim(xmin=0)
plt.ylim(ymin=0)
plt.ylabel('probability density')
plt.xlabel('output counts / gain')
plt.legend(legend_str)
plt.show()
